### ISSUE TITLE
Canonical version of these functions?

### ISSUE THREAD

**Issue by kamalmarhubi** [pubref/rules_require#1]
Hi @pcj! I am enjoying looking around at your Bazel rules projects, and like how you handle dependencies. I noted that the code in here is inlined in multiple repos. Is there one you consider canonical?

**Comment by pcj** [comment]
Not exactly.  This repo contains both the require.bzl function and stuff for transitive java dependencies.

I started down the transitive dependencies road via looking at https://github.com/pgr0ss/bazel-deps and rewriting some of that code.  Eventually I decided that approach was duplicating code that was more robust in tools like gradle and pivoted to the approach used in rules_maven (generating a build.gradle file and parsing the output).

You are right though, I've inlined the `require.bzl` functionality in several repos and have thought about canonicalizing it.
